### ISSUE TITLE
TypeError: Cannot read property 'destroy' of undefined 

### ISSUE THREAD

**Issue by bielas** [thevladeffect/ngx-typed-js#21]
Just copy and paste the example from docs and get an error like below:

```
TypeError: Cannot read property 'destroy' of undefined
    at NgxTypedJsComponent.ngOnChanges
```

```
	<ngx-typed-js [strings]="['Test!']" [loop]="true">
		<h1 class="typing"></h1>
	</ngx-typed-js>
```

How can you help me?

**Comment by kylanee** [comment]
I'm facing this issue too, after updating my project to Angular 12.

**Comment by steven274** [comment]
In Angular 12, ngOnChanges call before ngAfterViewInit

**Comment by ubaggili** [comment]
@steven274 I ran into the same issue and the fix was to swap the two as I believe you suggested.  I made the fix directly in the node_modules file as stop gap to test it out

Line 90 in 
from:
    ngOnChanges(changes) {
        this.typed.destroy();   // <-- this line needs to run after the ngAfterViewInit
        this.ngAfterViewInit();
    }

to
    ngOnChanges(changes) {
        this.ngAfterViewInit();
        this.typed.destroy();   // <-- swapped
    }

**Comment by thevladeffect** [comment]
This should be fixed in 2.1.1
